ddtrace/settings/asm.py

```diff
@@ -0,0 +1,50 @@
+from envier import Env
+
+from ddtrace.appsec._constants import API_SECURITY
+from ddtrace.appsec._constants import APPSEC
+from ddtrace.appsec._constants import DEFAULT
+from ddtrace.constants import APPSEC_ENV
+from ddtrace.constants import IAST_ENV
+
+
+def _validate_sample_rate(r: float) -> None:
+    if r < 0.0 or r > 1.0:
+        raise ValueError("sample rate value must be between 0.0 and 1.0")
+
+
+class ASMConfig(Env):
```

Comment:
why not just move this into an internal module, or `ddtrace/appsec/_settings.py` or something rather than making all these properties private?

in a world where we might move appsec code into it's own package, having the settings be within `ddtrace/appsec` could make sense to me.

Reply:
Tahir asked me to do it like dynamic_instrumentation did. I can change it, of course.

Reply:
aren't (at least some) of these variables public (I'm going by the assumption that a `_` prefix would denote private)? If so I think this is a good place, and perhaps we don't need to make them private here?

Reply:
I kept them private as they already were. We can always make them public. asm_enabled for example is probably something that could become public. But it would probably give the feeling to the users that you just need to toggle the value of some of them to activate or deactivate some feature at any time, while usually it can be a little more complex than that.